[abandoned][export] Store the arguments used to trace the exported program in itself

angelayi opened the pull request:
Summary:
X-link: https://github.com/pytorch/executorch/pull/115

Continuation of https://github.com/pytorch/pytorch/pull/107704

Test Plan: CI

Differential Revision: D48637348

angelayi commented:
cc @qihqi

avikchaudhuri commented on torch/_export/serde/serialize.py:
```diff
@@ -242,30 +242,27 @@ def deserialize_signature(sig: GraphSignature) -> ep.ExportGraphSignature:
 
 
-def serialize_state_dict(state_dict: Dict[str, Any]) -> bytes:
+def serialize_torch_artifact(artifact: Union[Dict[str, Any], Iterable[Any]]) -> bytes:
```
Is there any danger that original args can be more types of objects than state dict, and thus harder to `torch.save`? Or do you think we have enough restrictions that this is safe and general?

angelayi replied:
I think...torch.save should be able to handle most things? If not then users can not store these args. Otherwise I think it will be too much work on our side to figure out how to serialize different arguments.

avikchaudhuri commented on test/export/test_serialize.py:
```diff
@@ -103,7 +103,7 @@ def forward(self, x):
         exported_module = export(MyModule(), (input,))
 
-        serialized, _ = ExportedProgramSerializer().serialize(exported_module)
+        serialized, *_ = ExportedProgramSerializer().serialize(exported_module)
```
I don't quite like this *_  because it makes it look like there are a variable number of remaining things. IMO at this point we should consider using a dataclass / named tuple and just extract from it (`result.serialized`).

avikchaudhuri reviewed (approved):
Looks mostly OK, see question and nit.

angelayi committed:
[export] Store the arguments used to trace the exported program in itself (#107877)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/107877

X-link: https://github.com/pytorch/executorch/pull/115

Continuation of https://github.com/pytorch/pytorch/pull/107704

Test Plan: CI

Differential Revision: D48637348

fbshipit-source-id: 07e06550f6f83699f2a562638788bce63b8724de
